pkg/volume/azure_dd/azure_dd.go

```diff
@@ -146,6 +148,45 @@ func (plugin *azureDataDiskPlugin) GetVolumeLimits() (map[string]int64, error) {
 		return nil, fmt.Errorf("Expected Azure cloudprovider, got %s", cloud.ProviderName())
 	}
 
+	instances, ok := cloud.Instances()
+	if !ok {
+		glog.Warningf("Failed to get instances from cloud provider")
+		return volumeLimits, nil
+	}
+
+	instanceType, err := instances.InstanceType(context.TODO(), plugin.host.GetNodeName())
+	if err != nil {
+		glog.Errorf("Failed to get instance type from Azure cloud provider")
+		return volumeLimits, nil
+	}
+
+	az, err := getCloud(plugin.host)
+	if err != nil {
+		glog.Errorf("failed to get azure cloud in GetVolumeLimits, plugin.host: %s", plugin.host.GetHostName())
+		return volumeLimits, nil
+	}
+
+	result, err := az.VirtualMachineSizesClient.List(context.TODO(), az.Location)
+	if err != nil || result.Value == nil {
+		glog.Errorf("failed to list vm sizes in GetVolumeLimits, plugin.host: %s, location: %s", plugin.host.GetHostName(), az.Location)
+		return volumeLimits, nil
+	}
+
+	vmsize := strings.ToUpper(instanceType)
+	for _, size := range *result.Value {
+		if size.Name == nil || size.MaxDataDiskCount == nil {
+			glog.Errorf("failed to get vm size in GetVolumeLimits, plugin.host: %s, location: %s", plugin.host.GetHostName(), az.Location)
+			continue
+		}
+		if strings.ToUpper(*size.Name) == vmsize {
+			glog.V(2).Infof("got a matching size in GetVolumeLimits, Name: %s, MaxDataDiskCount: %s", *size.Name, *size.MaxDataDiskCount)
+			volumeLimits = map[string]int64{
+				util.AzureVolumeLimitKey: int64(*size.MaxDataDiskCount),
+			}
+			break
+		}
```

Comment:
can we write some tests for this? e2e will be obviously great but some unit tests can be added for this particular function?

Reply:
added a unit test in new commit